apps/dog-food/ChatTerminal.hs

```diff
@@ -0,0 +1,321 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module ChatTerminal
+  ( ChatTerminal (..),
+    newChatTerminal,
+    chatTerminal,
+    updateUsername,
+    ttyContact,
+    ttyFromContact,
+  )
+where
+
+import Control.Concurrent (threadDelay)
+import Control.Concurrent.Async (race_)
+import Control.Concurrent.STM
+import Control.Monad
+import Data.ByteString.Char8 (ByteString)
+import qualified Data.ByteString.Char8 as B
+import Data.List (dropWhileEnd)
+import Data.Maybe (fromMaybe)
+import qualified Data.Text as T
+import Data.Text.Encoding
+import Numeric.Natural
+import Simplex.Messaging.Transport (getLn, putLn)
+import qualified System.Console.ANSI as C
+import System.IO
+import Types
+
+data ChatTerminal = ChatTerminal
+  { inputQ :: TBQueue ByteString,
+    outputQ :: TBQueue ByteString,
+    activeContact :: TVar (Maybe Contact),
+    username :: TVar (Maybe Contact),
```

Comment:
might not be required currently, but why not have a separate `User` type? Wouldn't there be possible differences between them? Even now when reading Main line 122 `let user = Contact <$> name` I initially wondered why is user a contact.. maybe `Contact` can be part of a `User` - I understand user also needs his contact for sharing in future

Reply:
I am still treating it as a throw-away thing...